### PR TITLE
[YAML] Support comment and delimiter attributes on CSV IO.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/yaml/standard_io.yaml [hunk 1]
```diff
@@ -139,8 +139,11 @@
     mappings:
       'ReadFromCsv':
         path: 'path'
+        delimiter: 'delimiter'
+        comment: 'comment'
```
Comment (johnjcasey) [inline sdks/python/apache_beam/yaml/standard_io.yaml:143]: I don't see any changes in the transform provider related to comment, is that intentional?

Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_io.yaml:143]: These parameters were in the upstream transform (along with everything else from Pandas); this change simply exposes them.

Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_io.yaml:143]: So, now that I had a chance to actually look at this, it is WAI. 

The Python Reader explicitly subsets (via the renaming transform) to comment and delimiter, while the writer supports delimiter (a comment marker specifying lines to ignore doesn't makes sense for a writer). 

There is no Java CSV reader. The writer supports a delimiter (but, as with Python, no need for comment). These are read directly from the provided transform config schema at discovery time. 

Sorry for the confusion.

Comment (johnjcasey) [reply sdks/python/apache_beam/yaml/standard_io.yaml:143]: SGTM, LGTM

Comment (johnjcasey) [reply sdks/python/apache_beam/yaml/standard_io.yaml:143]: SGTM, LGTM

**Review by johnjcasey — APPROVED** [review]

**Pull request merged**